iroh-gossip/src/proto/util.rs

```diff
@@ -0,0 +1,185 @@
+//! Utilities used in the protocol implementation
+
+use rand::{
+    seq::{IteratorRandom, SliceRandom},
+    Rng,
+};
+use std::{
+    collections::BTreeMap,
+    hash::Hash,
+    time::{Duration, Instant},
+};
+
+/// A hash set where the iteration order of the values is independent of their
+/// hash values.
+///
+/// This is wrapper around [indexmap::IndexSet] that limits the removal API to
+/// always do shift_remove (preserving the order of other elements) and adds a
+/// couple of utility methods to randomly select elements from the set.
+#[derive(Debug, Clone)]
```

Comment:
I thought about this too and wasn't sure because a general tendency seems to be to not use `Deref` too much but actually it would make sense here and be fine I think now.